cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/ReadSegmentRangeCommand.java

```diff
@@ -42,29 +45,43 @@ public ReadSegmentRangeCommand(CommandArgs args) {
     }
 
     @Override
-    public void execute() throws ExecutionException, InterruptedException, TimeoutException {
-        ensureArgCount(4);
+    public void execute() throws ExecutionException, InterruptedException, TimeoutException, IOException {
+        ensureArgCount(5);
 
         final String fullyQualifiedSegmentName = getArg(0);
         final int offset = getIntArg(1);
         final int length = getIntArg(2);
         final String segmentStoreHost = getArg(3);
+        final String fileName = getArg(4);
         @Cleanup
         CuratorFramework zkClient = createZKClient();
         @Cleanup
         SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient);
         CompletableFuture<WireCommands.SegmentRead> reply = segmentHelper.readSegment(fullyQualifiedSegmentName,
                 offset, length, new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), "");
         WireCommands.SegmentRead segmentRead = reply.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
-        output("ReadSegment: %s", segmentRead.toString());
-        output("SegmentRead content: %s", segmentRead.getData().toString(StandardCharsets.UTF_8));
+        File f = new File(fileName);
+        // If file exists throw FileAlreadyExistsException, an existing file should not be overwritten with new data.
+        if (f.exists()) {
+            throw new FileAlreadyExistsException("Cannot write segment data into a file that already exists.");
+        }
+        if (!f.getParentFile().exists()) {
+            f.getParentFile().mkdirs();
+        }
+        f.createNewFile();
+        // Write data into the file.
+        @Cleanup
+        FileOutputStream fileStream = new FileOutputStream(f);
+        fileStream.write(segmentRead.getData().array());
```

Comment:
`.array()` will give you the array over which this `ArrayView/ByteArraySegment` is built upon, however that may be more than needed. An `ArrayView` is a slice of a `byte[]`, so it has a start offset and length. You are therefore possibly writing more than requested (you may include other unrelated stuff that's on that array).

Reply:
Instead of `.array()` I use `.readBytes()` to directly write it into the `FileOutputStream`.